wrappers/python/openmm/app/pdbreporter.py

```diff
@@ -91,15 +97,81 @@ def report(self, simulation, state):
         state : State
             The current state of the simulation
         """
+        if self._atomSubset is not None:
+            if self._subsetTopology is None:
+                self._createSubsetTopology(simulation.topology)
+
+            topology = self._subsetTopology
+
+            #PDBFile will convert to angstroms so do it here first instead
+            positions = state.getPositions().value_in_unit(angstroms) 
+            positions = [positions[i] for i in self._atomSubset]
+
+        else:
+            topology = simulation.topology
+            positions = state.getPositions()
+
         if self._nextModel == 0:
-            PDBFile.writeHeader(simulation.topology, self._out)
-            self._topology = simulation.topology
+            PDBFile.writeHeader(topology, self._out)
+            self._topology = topology
             self._nextModel += 1
-        PDBFile.writeModel(simulation.topology, state.getPositions(), self._out, self._nextModel)
+        PDBFile.writeModel(topology, positions, self._out, self._nextModel)
         self._nextModel += 1
         if hasattr(self._out, 'flush') and callable(self._out.flush):
             self._out.flush()
 
+    def _createSubsetTopology(self, topology):
+        """Create a subset of an existing topology.
+
+        Parameters
+        ----------
+        topology : Topology
+            The Topology to create a subset from
+        """
+        # check atomSubset is valid 
+        if len(self._atomSubset)==0:
+            raise ValueError('atomSubset cannot be an empty list')
+        if not all(a==int(a) for a in self._atomSubset):
+            raise ValueError('all of the indices in atomSubset must be integers')
+        if len(set(self._atomSubset)) != len(self._atomSubset):
+            raise ValueError('atomSubset must contain unique indices')
+        if sorted(self._atomSubset) != self._atomSubset:
+            raise ValueError('atomSubset must be sorted in ascending order')
+        if self._atomSubset[0] < 0:
+            raise ValueError('The smallest allowed value in atomSubset is zero')
+        if self._atomSubset[-1] >= topology.getNumAtoms():
+            raise ValueError('The maximum allowed value in atomSubset must be less than the total number of particles')
+        
+
+        self._subsetTopology = Topology()
+        
+        # convert to set for fast look up
+        atomSubsetSet=set(self._atomSubset)
+
+        # store a map from posIndex to Atom object for when we add the bonds
+        indexToAtom = {}
+
+        posIndex = 0
+        for chain in topology.chains():
+            c = self._subsetTopology.addChain(chain.id)
+            residues = list(chain.residues())
+            for res in residues:
+                r = self._subsetTopology.addResidue(res.name,c,res.id,res.insertionCode)
```

Comment:
Can you make the formatting consistent?  Spaces after the commas.  There are also a few places where you don't put spaces around `=` in assignments.